src/main/java/hudson/remoting/Launcher.java

```diff
@@ -491,19 +636,7 @@ public List<String> parseJnlpArguments() throws ParserConfigurationException, SA
         while (true) {
             URLConnection con = null;
             try {
-                con = Util.openURLConnection(agentJnlpURL);
-                if (con instanceof HttpURLConnection) {
-                    HttpURLConnection http = (HttpURLConnection) con;
-                    if  (agentJnlpCredentials != null) {
-                        String userPassword = agentJnlpCredentials;
-                        String encoding = Base64.getEncoder().encodeToString(userPassword.getBytes(StandardCharsets.UTF_8));
-                        http.setRequestProperty("Authorization", "Basic " + encoding);
-                    }
-                    if (System.getProperty("proxyCredentials", proxyCredentials) != null) {
-                        String encoding = Base64.getEncoder().encodeToString(System.getProperty("proxyCredentials", proxyCredentials).getBytes(StandardCharsets.UTF_8));
-                        http.setRequestProperty("Proxy-Authorization", "Basic " + encoding);
-                    }
-                }
```

Comment:
Deduplicating this logic, which also lived in the method that is now being called. Also note that this was the real logic that handled authentication, not the dead code above that was used for the `-auth` argument.